### ISSUE TITLE
box_version not working in multi-machine configuration

### ISSUE THREAD

**Issue by LukasVice** [mitchellh/vagrant#4794]
this config works as expected:

```
Vagrant.configure("2") do |config|
    config.vm.box_version = "> 2.0"
...
```

but this does not work:

```
Vagrant.configure("2") do |config|
    config.vm.define "mybox", primary: true do |mybox|
        mybox.vm.box_version = "> 2.0"
...
```

or am i doing something wrong?

thanks

**Comment by sethvargo** [comment]
@LukasVice this is the correct behavior. If the latest version of the box is 2.0, `>= 2.0` is valid, and 2.0 is the latest version for the constraint `>= 1.0`.

**Comment by LukasVice** [comment]
Sorry, but i do not understand this. :/ If my local box version is 1.0 but there is a box available with version 2.0 and the Vagrantfile config says box_version ">= 2.0"... shouldn't the new box be downloaded?
